### PR TITLE
Incorporates feedback from PR #645

### PR DESCRIPTION
**Pull request by zslayton**
Builds on #645 and #647.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -59,7 +59,7 @@ bytes = "0.4"
 # this makes it explicitly not do this as there is an advisory warning against this:
 # See: https://github.com/chronotope/chrono/issues/602
 chrono = { version = "0.4", default-features = false, features = ["clock", "std", "wasmbind"] }
-delegate = "0.9.0"
+delegate = "0.10.0"
```
Comment (zslayton) [inline Cargo.toml:62]: 🗺️ Bumped to the latest release.

File: src/lazy/encoding.rs [hunk 2]
```diff
@@ -7,33 +7,64 @@ use crate::lazy::binary::raw::reader::LazyRawBinaryReader;
...
+    LazyRawTextValue, LazyRawTextValue_1_0, LazyRawTextValue_1_1, MatchedRawTextValue,
+    RawTextAnnotationsIterator,
 };
-use crate::IonResult;
+use std::fmt::Debug;
+
+/// Marker trait for types that represent an Ion encoding.
+pub trait Encoding: Debug + Copy {
+    fn name() -> &'static str;
+}
```
Comment (zslayton) [inline src/lazy/encoding.rs:27]: 🗺️ I added this partly to have a generic implementation of `Debug` for `RawTextValue<'data, E: TextEncoding>`.

File: src/lazy/encoding.rs [hunk 3]
```diff
@@ -54,20 +85,9 @@ impl<'data> TextEncoding<'data> for TextEncoding_1_1 {
     }
 }
 
-/// The Ion 1.0 binary encoding.
```
Comment (zslayton) [inline src/lazy/encoding.rs:57]: 🗺️ I moved things around in this file to make read a bit better from top to bottom. Traits and their impls are grouped throughout.

File: src/lazy/expanded/macro_table.rs [hunk 4]
```diff
@@ -5,13 +5,23 @@ use std::collections::HashMap;
...
 
+impl MacroKind {
+    fn name(&self) -> &str {
+        match self {
+            MacroKind::Void => "void",
+            MacroKind::Values => "values",
+            MacroKind::MakeString => "make_string",
+        }
+    }
+}
```
Comment (zslayton) [inline src/lazy/expanded/macro_table.rs:23]: 🗺️ Per feedback, I added this method and used it to initialize the `MacroTable`.

File: src/lazy/never.rs
Comment (zslayton) [inline src/lazy/never.rs:1]: 🗺️ This type was previously in the `encoding.rs` file, now it's its own module.

**Review by zslayton — COMMENTED** [review]
🗺️ PR tour

File: src/lazy/text/value.rs [hunk 5]
```diff
@@ -40,41 +41,47 @@ impl<'a> Debug for MatchedRawTextValue<'a> {
 // ===== Version-specific wrappers =====
 //
 // These types provide Ion-version-specific impls of the LazyRawValue trait
-#[derive(Copy, Clone, Debug)]
-pub struct LazyRawTextValue_1_0<'data> {
+#[derive(Copy, Clone)]
+pub struct LazyRawTextValue<'data, E: TextEncoding<'data> + Copy> {
     pub(crate) matched: MatchedRawTextValue<'data>,
+    spooky: PhantomData<E>,
 }
```
Comment (zslayton) [inline src/lazy/text/value.rs:48]: 🗺️ Changes in this file stem from this change, which makes `LazyRawTextValue` generic over the text encoding version. Below, you can see that `LazyRawTextValue_1_0` and `LazyRawTextValue_1_1` are now just type aliases.

File: src/lazy/text/value.rs [hunk 6]
```diff
@@ -145,51 +149,27 @@ impl<'data, D: TextEncoding<'data>> LazyRawValue<'data, D> for MatchedRawTextVal
     }
 }
 
-impl<'data> LazyRawValuePrivate<'data> for LazyRawTextValue_1_0<'data> {
```
Comment (zslayton) [inline src/lazy/text/value.rs]: 🗺️ Now that `*_1_0` and `*_1_1` share an implementation, I was able to delete this redundant `impl`.

**Review by popematt — APPROVED** [review]

**Pull request merged**